t/ui/27-plugin_obs_rsync_status_details.t

```diff
@@ -50,21 +50,30 @@ my $host = "http://127.0.0.1:$port";
 sub fake_api_server {
     my $mock = Mojolicious->new;
     $mock->mode('test');
-    $mock->routes->get(
-        '/public/build/:proj/_result' => sub {
-            my $c     = shift;
-            my $proj  = $c->stash('proj');
-            my %repos = (
-                'Proj1'       => 'standard',
-                'BatchedProj' => 'containers',
-            );
-            my $repo = $repos{$proj};
-            $repo = 'images' unless $repo;
-            $c->render(
-                status => 200,
-                text => qq{<result project="$proj" repository="$repo" arch="local" code="published" state="published">}
-            );
-        });
+
+    my $sub = sub {
+        my $c       = shift;
+        my $proj    = $c->stash('proj');
+        my $package = $c->stash('package') // '';
+        $c->render(
+            status => 404,
+            test   => 'unknown package'
+        ) if $proj eq 'Proj2' && $package ne 'mypackage';
+        my %repos = (
+            'Proj1'       => 'standard',
+            'Proj2'       => 'appliances',
+            'BatchedProj' => 'containers',
+        );
+        my $repo = $repos{$proj};
+        $repo = 'images' unless $repo;
+        $c->render(
+            status => 200,
+            text   => qq{<result project="$proj" repository="$repo" arch="local" code="published" state="published">
+<result project="$proj" repository="images" arch="local" code="building" state="building">}
+        );
+    };
+    $mock->routes->get('/public/build/:proj/_result'                  => \&$sub);
```

Comment:
Can't you just pass the `$sub`variable normally? I'd rather avoid this scary syntax.